### ISSUE TITLE
Does not work when loaded as an ESM module

### ISSUE THREAD

**Issue by kirlat** [johnagan/clean-webpack-plugin#84]
The plugin works great when loaded as `require('clean-webpack-plugin')`. However, we need to run it in a node.js build script that uses an ESM import syntax, with `--experimental-modules` flag (node.js 10). This is how we load it: `import CleanWebpackPlugin from 'clean-webpack-plugin'`. Unfortunately, it returns the following error: `TypeError: Cannot read property 'filename' of undefined`. I've checked the plugin code, and the only place where `filename` is used is `module.parent.filename`. I checked, and `module.parent` is undefined when plugin is loaded using ESM syntax.

What would be the best way to fix this? It seems that `module.parent.filename` is used to determine Webpack location. However, plugin takes a `root` option that is a Webpack root path. Can we check if module.parent is undefined and if it is, use a `root` option instead of it?

This plugin is really helpful and I would love to be able to use it in an ESM environment.

**Comment by i8degrees** [comment]
Hi, I just wanted to throw it out there that I suffer the same problem as @kirlat above. I use `@std/esm` for an ESM environment and decided to take a stab at resolving the issue with using the suggestion made. I forget now why exactly I was not able to pass the existing `root` option as-is, so I added an additional option under the name of `context`.

```
new CleanWebpackPlugin(['dist'], {
      root: __dirname,
      context: __dirname,
    })
```

I have been using this successfully for the past few months now without hiccup. Here's to hoping that somebody with more time might be able to nail this problem in its coffin! :-)

Thanks!

**Comment by riri** [comment]
I got exactly the same problem than @kirlat, using ESM, `TypeError: Cannot read property 'filename' of undefined`, as `module.parent` is undefined.

**Comment by MeiKatz** [comment]
I am having the same problem currently and I think the solution is to make the check against `module.parent` conditionally: if it is not defined then `webpackDir` should be `undefined`.
```javascript
webpackDir = module.parent && path.dirname(module.parent.filename);
```
And so the problem is solved.

**Comment by chrisblossom** [comment]
Closed via #95 for `^1.0.0`.

[`v2.0.0`](https://github.com/johnagan/clean-webpack-plugin/tree/v2.0.0) via #99 has been released. See [johnagan/clean-webpack-plugin#usage](https://github.com/johnagan/clean-webpack-plugin#usage). Please open a new issue if you continue to have problems.
